### PR TITLE
fix: Stop raising errors for unsupported Whatsapp messages

### LINKED ISSUE #5539: ArgumentError: bad argument (expected URI object or URI string)
Sentry Issue: [CHATWOOT-NN](https://sentry.io/organizations/chatwoot-p3/issues/3435111000/?referrer=github_integration)

```
ArgumentError: bad argument (expected URI object or URI string)
  app/services/whatsapp/incoming_message_whatsapp_cloud_service.rb:13:in `download_attachment_file'
    Down.download(url_response.parsed_response['url'], headers: inbox.channel.api_headers)
  app/services/whatsapp/incoming_message_base_service.rb:93:in `attach_files'
    attachment_file = download_attachment_file(attachment_payload)
  app/services/whatsapp/incoming_message_base_service.rb:25:in `perform'
    attach_files
  app/jobs/webhooks/whatsapp_events_job.rb:10:in `perform'
    Whatsapp::IncomingMessageWhatsappCloudService.new(inbox: channel.inbox, params: params).perform
...
(84 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Save the message when `message_type: reaction`

Fixes #5539 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested on local and added the test cases for the same.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
@sojan-official I have made the respective changes as discussed.
Added the test cases to prove that we are ignoring reaction-type messages as well as not raising any errors for the same.

**Review by sojan-official — CHANGES_REQUESTED** [review]

File: spec/jobs/webhooks/whatsapp_events_job_spec.rb [hunk 1]
```diff
@@ -62,6 +62,33 @@
...
+                display_phone_number: other_channel.phone_number.delete('+')
+              }
+            }
+          }]
+        }]
+      }.with_indifferent_access
+      expect do
+        Whatsapp::IncomingMessageWhatsappCloudService.new(inbox: other_channel.inbox, params: wb_params).perform
+      end.not_to raise_error 'ArgumentError'
+      expect(Message.last.content).to be_nil
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply spec/jobs/webhooks/whatsapp_events_job_spec.rb]: @sojan-official Made the respective changes.

**Review by muhsin-k — DISMISSED** [review]

File: app/services/whatsapp/incoming_message_base_service.rb [hunk 2]
```diff
@@ -86,8 +86,12 @@ def message_type
     @processed_params[:messages].first[:type]
   end
 
+  def unprocessable_message_type?
+    %w[reaction].include?(message_type)
+  end
+
   def attach_files
-    return if %w[text button interactive].include?(message_type)
+    return if %w[text button interactive reaction].include?(message_type)
```
Comment (sojan-official) [inline app/services/whatsapp/incoming_message_base_service.rb:94]: do we need this check ?

**Comment by tejaswinichile** [comment]
Note: We will be taking care of https://sentry.io/share/issue/128633bcfc274c07828021dd3d28933e/ this issue in this current change.

**Pull request merged**